pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -159,11 +164,41 @@ func (c *HPAWatcherClient) Start() {
 				// Updating the metrics against Datadog should not affect the HPA pipeline.
 				// If metrics are temporarily unavailable for too long, they will become `Valid=false` and won't be evaluated.
 				c.updateExternalMetrics()
+			case <-gcTicker.C:
+				if !leaderEngine.IsLeader() {
```

Comment:
How confident are we that the local stores across replicas are consistent ?
If the leader updates the shared store (cm) but one of the replicas is not yet up to date with the previous state, will that work as expected ? Are we in an eventually consistent pattern ?